src/components/common/SearchBar/index.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { ChangeEvent, InputHTMLAttributes, useState } from 'react';
+
+import SearchIcon from '/public/SearchIcon.svg';
+import DeleteIcon from '/public/DeleteIcon.svg';
+
+import Input from '../Input';
+
+interface SearchBarProps extends InputHTMLAttributes<HTMLInputElement> {
+  icon?: React.ReactNode;
+}
+
+export default function SearchBar({ icon }: SearchBarProps) {
+  const [inputValue, setInputValue] = useState('');
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  const clearInput = () => {
+    setInputValue('');
+  };
+
+  let iconElement = null;
+  if (inputValue) {
+    iconElement = (
+      <div onClick={clearInput} className="cursor-pointer">
+        <DeleteIcon />
+      </div>
+    );
+  } else {
+    iconElement = (
+      <div>
+        <SearchIcon />
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <div className="flex items-center px-4 py-2">
+        <div className="w-[40px] h-[40px]">{icon}</div>
+        <Input
+          rightItem={iconElement}
+          onChange={handleChange}
+          onClick={clearInput}
+          value={inputValue}
+        />
+      </div>
+    </div>
+  );
```

Comment:
모던 자바스크립트에서는 let 사용을 최소화하는 것이 트렌드인 것 같습니다
이유는 작성해주신 로직처럼 let으로 선언된 변수는 재할당이 가능하기 때문이라고 생각해요.
프로그램에서 변경의 추적이 어려워지거나, 예상치못한 값의 변경은 지양하는 것이 좋지 않을까~ 생각합니다.

작성해주신 로직을 수정해보면...?

```suggestion
  return (
    <div>
      <div className="flex items-center px-4 py-2">
        <div className="w-[40px] h-[40px]">{icon}</div>
        <Input
          rightItem={text ? <button onClick={resetText}><DeleteIcon /></ button> : <SearchIcon />}
          value={text}
          onChange={onTextChange}
        />
      </div>
    </div>
  );
```

요렇게 수정할 수 있을 것 같아요!
삼항연산자를 사용해서 입력된 값이 있을때는 삭제버튼을, 입력된 값이 없을때는 돋보기 아이콘을 보여줄 수 있어요.

Reply:
https://github.com/depromeet/ddoeat_client/pull/37/commits/a727b8ef12febfa608797d834abc25b7ff6d3793 에서 리뷰 반영했습니다! 감사합니다~~:)